go/vt/mysqlctl/replication.go

```diff
@@ -819,3 +819,31 @@ func (mysqld *Mysqld) SemiSyncExtensionLoaded(ctx context.Context) (mysql.SemiSy
 
 	return conn.Conn.SemiSyncExtensionLoaded()
 }
+
+func (mysqld *Mysqld) IsSemiSyncBlocked(ctx context.Context) (bool, error) {
+	conn, err := getPoolReconnect(ctx, mysqld.dbaPool)
```

Comment:
I would not expect that we need to use the DBA pool for this as we're just selecting from P_S. The DBA pools tend to be relatively small and should only be used when truly necessary or they can block other things that are critical and do require it.

Reply:
I think I used this pool because that's what all the other operations (like `SetReplicationSource`) do too:

https://github.com/vitessio/vitess/blob/429964ed2f4959e4e903ac09684c2dae2aca2687/go/vt/mysqlctl/replication.go#L496-L506

Maybe it would be overall better to fetch a connection from the pool and then use that connection for all the operations that happen during commands like `demotePrimary`, but that's a bigger change that's outside of the scope of this PR.

Reply:
That's fair. I don't think we *should* use DBA everywhere here, but it does line up with the other uses. And using a single connection for all of the work might be a good idea here too. Not related though so we can investigate this more separately.